parquet-hadoop/src/main/java/org/apache/parquet/hadoop/rewrite/RewriteOptions.java

```diff
@@ -245,6 +283,31 @@ public Builder(Configuration conf, List<Path> inputFiles, Path outputFile) {
     public Builder(ParquetConfiguration conf, List<InputFile> inputFiles, OutputFile outputFile) {
       this.conf = conf;
       this.inputFiles = inputFiles;
+      this.inputFilesToJoin = new ArrayList<>();
+      this.outputFile = outputFile;
+    }
+
+    public Builder(Configuration conf, List<Path> inputFiles, List<Path> inputFilesToJoin, Path outputFile) {
+      this.conf = new HadoopParquetConfiguration(conf);
+      this.inputFiles = new ArrayList<>(inputFiles.size());
+      for (Path inputFile : inputFiles) {
+        this.inputFiles.add(HadoopInputFile.fromPathUnchecked(inputFile, conf));
+      }
+      this.inputFilesToJoin = new ArrayList<>(inputFilesToJoin.size());
+      for (Path inputFile : inputFilesToJoin) {
```

Comment:
Do we support both rewrite and join?

Reply:
You mean over-writing existing column and joining a new column? If so, yes.